corehq/apps/api/resources/v0_5.py

```diff
@@ -1074,12 +1074,27 @@ class NavigationEventAuditResourceParams:
 
     def __post_init__(self, raw_params=None):
         if raw_params:
+            self._validate_keys(raw_params)
+
             self._set_compound_keys(raw_params)
             self.users = raw_params.getlist('users')
             self.limit = raw_params.get('limit')
             self.local_timezone = raw_params.get('local_timezone')
             self.cursor = raw_params.get('cursor')
 
+    def _validate_keys(self, params):
+        valid_keys = {'users', 'limit', 'local_timezone', 'cursor', 'format', 'local_date'}
```

Comment:
Nit: you can also avoid repeating the field names by doing something like:

`{field.name for field in dataclasses.fields(self)}`

Reply:
I was thinking of doing that originally but that would also include `cursor_local_date` and `cursor_user` which aren't valid keys.  I don't think it would be better to have to add a step to remove them from `{field.name for field in dataclasses.fields(self)}`.